plugins/core/profile.go

```diff
@@ -0,0 +1,343 @@
+			task.SerialNumber = arg.Value
+		}
+	}
+	fmt.Println("adding profile task:", task)
+	if _, exists := m.TraceProfileTasks[task.TaskId]; exists {
+		return
+	}
+	endTime := task.StartTime + int64(task.Duration)*60*1000
+	task.EndTime = endTime
+	task.Status = reporter.Pending
+	m.TraceProfileTasks[task.TaskId] = &task
+}
+
+func (m *ProfileManager) RemoveProfileTask() {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	for k, t := range m.TraceProfileTasks {
+		if t.Status == reporter.Reported || t.EndTime < time.Now().Unix() {
+			delete(m.TraceProfileTasks, k)
+		}
+	}
+}
+
+func (m *ProfileManager) getProfileTask(endpoint string) []*reporter.TraceProfileTask {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	var result []*reporter.TraceProfileTask
+	for _, t := range m.TraceProfileTasks {
+		endTime := t.StartTime + int64(t.Duration)*60*1000
+		if t.EndpointName == endpoint && t.StartTime <= time.Now().UnixMilli() && endTime > time.Now().UnixMilli() && t.Status == reporter.Pending {
+			result = append(result, t)
+		}
+	}
+	return result
+}
+
+func (m *ProfileManager) IfProfiling() bool {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	return m.status
+}
+
+func (m *ProfileManager) generateProfileLabels(traceSegmentID string, minDurationThreshold int64) profileLabels {
+	var l = &LabelSet{}
+	if minDurationThreshold == 0 {
+		l = Labels(l, SegmentLabel, traceSegmentID)
+	} else {
+		l = Labels(l, SegmentLabel, traceSegmentID, MinDurationLabel, strconv.FormatInt(minDurationThreshold, 10))
+	}
+	closeChan := make(chan struct{}, 1)
+	return profileLabels{
+		labels:    l,
+		closeChan: closeChan,
+	}
+}
+
+func (m *ProfileManager) generateCurrentTask(t *reporter.TraceProfileTask, traceSegmentID string) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	var c = currentTask{
+		serialNumber:         t.SerialNumber,
+		taskId:               t.TaskId,
+		traceSegmentId:       traceSegmentID,
+		minDurationThreshold: t.MinDurationThreshold,
+		duration:             t.Duration,
+	}
+	m.currentTask = &c
+}
+
+func (m *ProfileManager) ToProfile(endpoint string, traceSegmentID string) {
+	//check if profiling
+	if m.IfProfiling() {
+		c := m.generateProfileLabels(traceSegmentID, 0)
+		m.labelSets[traceSegmentID] = c
+		SetGoroutineLabels(c.labels)
+		return
+	}
+
+	tasks := m.getProfileTask(endpoint)
+	if tasks != nil {
+		for _, v := range tasks {
+			m.TraceProfileTasks[v.TaskId].Status = reporter.Running
+			//choose task to profiling
+			task := v
+			m.generateCurrentTask(task, traceSegmentID)
+			err := m.StartProfiling(traceSegmentID)
+			if err != nil {
+				fmt.Println(err)
+				return
+			}
+			go func(task *reporter.TraceProfileTask) {
+				err = m.monitor()
+				if err != nil {
+					m.TraceProfileTasks[task.TaskId].Status = reporter.Pending
+					m.currentTask = nil
+					m.status = false
+					return
+				}
+				m.TraceProfileTasks[task.TaskId].Status = reporter.Finished
+			}(task)
+			break
+		}
+	}
+
+}
+
+func (m *ProfileManager) StartProfiling(traceSegmentID string) error {
```

Comment:
CPU Profiling should be a global work, not related to a specific trace. Because the endpoint will be executing. 
Such as there is one endpoint that receives two identical requests at the same time, `pprof.StartCPUProfile` will be executed twice.